app/util/autoversioning.py

```diff
@@ -80,26 +92,19 @@ def _calculate_source_version():
     this will *not* work from a frozen package (which does not include the git repo data). This will only work in the
     context of running the application from the cloned git repo.
 
-    If one of the git commands used to calculate the version fails unexpectedly, the patch in the version string will
-    be set to "???".
```

Comment:
Can you add a comment that if this is run outside of a git repo that it will raise `subprocess.CalledProcessError`?

Alternatively you could keep the try/except and just make this method return `None` if it fails. That would reduce coupling between the calling code and the implementation of this method.